asserts/snapasserts/validation_sets.go

```diff
@@ -281,3 +328,84 @@ func (v *ValidationSets) Conflict() error {
 	}
 	return nil
 }
+
+type snapRef struct {
+	naming.SnapRef
+	revision snap.Revision
+}
+
+// CheckInstalledSnaps checks installed snaps against the validation sets.
+func (v *ValidationSets) CheckInstalledSnaps(snaps []*InstalledSnap) error {
+	installed := naming.NewSnapSet(nil)
+	for _, sn := range snaps {
+		installed.Add(sn)
+	}
+
+	// snapName -> validationSet key -> validation set
+	invalid := make(map[string]map[string]bool)
+	missing := make(map[string]map[string]bool)
+	wrongrev := make(map[string]map[string]bool)
+	sets := make(map[string]*asserts.ValidationSet)
+
+	for _, cstrs := range v.snaps {
+		for rev, revCstr := range cstrs.revisions {
+			for _, rc := range revCstr {
+				ref := naming.NewSnapRef(rc.Name, rc.SnapID)
```

Comment:
I think rc is or should be a SnapRef already because asserts.ValidationSetSnap is one

Reply:
Indeed! Thanks